src/codeu/chat/server/MessagePersist.java

```diff
@@ -0,0 +1,182 @@
+import codeu.chat.common.RawController;
+import codeu.chat.common.User;
+import codeu.chat.util.Logger;
+import codeu.chat.util.Time;
+import codeu.chat.util.Uuid;
+
+public final class MessagePersist {
+
+	private Connection conn;
+	private Statement stmt;
+
+	public MessagePersist() {
+		try {
+			Class.forName("org.sqlite.JDBC");
+			conn = DriverManager.getConnection("jdbc:sqlite:message.db");
+		}
+		catch (Exception e) {
+			System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+    		System.exit(0);
+		}
+	}
+
+	public void createUserTable() {
+		try {
+			stmt = conn.createStatement();
+			String sql = "CREATE TABLE IF NOT EXISTS USERS " + 
+						 "(ID            TEXT      NOT NULL," + 
+						 " NAME          TEXT                  NOT NULL," + 
+						 " CREATIONTIME  INTEGER            NOT NULL)";
+			stmt.executeUpdate(sql);
+			stmt.close();
+		}
+		catch (Exception e) {
+			System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+    		System.exit(0);
+    	}
+	}
+
+	public void insertUserTable(String id, String name, Long time) {
+		try {
+			String sql = "INSERT INTO USERS (ID,NAME,CREATIONTIME) " + 
+						 "VALUES ( '" + id + "' , '" + name + "' , '" + time + "');" ;
+			stmt.executeUpdate(sql);
+			stmt.close();
+		}
+		catch (Exception e) {
+			System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+      		System.exit(0);			
+		}
+	}
+
+	public void retrieveUser(Controller controller) {
+		try {
+			ResultSet rs = stmt.executeQuery( "SELECT * FROM USERS;");
+			while (rs.next() ) {
+				Uuid id = Uuid.fromString(rs.getString("id"));
+				String name = rs.getString("name");
+				Time time = Time.fromMs(rs.getLong("creationtime"));
+
+				controller.newUser(id, name, time);
+			}
+			rs.close();
+			stmt.close();
+		}
+		catch ( Exception e ) {
+      		System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+      		System.exit(0);
+   		}
+	}
+
+	public void createConversationTable() {
+		try {
+			String sql = "CREATE TABLE IF NOT EXISTS CONVERSATION " + 
+						 "(ID            TEXT      NOT NULL," + 
+						 " TITLE         TEXT      NOT NULL," + 
+						 " OWNER         TEXT      NOT NULL," + 
+						 " CREATIONTIME  INTEGER      NOT NULL)";
+			stmt.executeUpdate(sql);
+			stmt.close();
+		}
+		catch (Exception e) {
+			System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+    		System.exit(0);
+    	}
+    }
+
+	public void insertConversationTable(String id, String title, String ownerID, Long time) {
+		try {
+			String sql = "INSERT INTO CONVERSATION (ID, TITLE, OWNER, CREATIONTIME)" + 
+						 "VALUES ( '" + id + "' , '" + title + "' , '" + ownerID + "' , '" + time + "');" ;
+			stmt.executeUpdate(sql);
+			stmt.close();
+		}
+		catch (Exception e) {
+			System.err.println( e.getClass().getName() + ": " + e.getMessage() );
+      		System.exit(0);			
+		}
+
+	}
+
+	public void retrieveConversation(Controller controller) {

```

Comment:
Some of the functions in this class are actually doing two things:
1. querying the database to get all records
2. initializing the Controller with those database records

This is a great learning opportunity about a very important software engineering concept called the Single Responsibility Principle. Classes and functions should specialize in what they do. https://en.wikipedia.org/wiki/Single_responsibility_principle